### ISSUE TITLE
Namespace Microsoft.Quantum.Simulation.QuantumProcessor.Extensions shouldn't be visible.

### ISSUE THREAD

**Issue by cgranade** [microsoft/qsharp-runtime#327]
## Describe the bug
The [`Microsoft.Quantum.Simulation.QuantumProcessor.Extensions` namespace](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.simulation.quantumprocessor.extensions) is an internal implementation detail to the `IQuantumProcessor` interface in the C# runtime, and not a public Q# API, but appears as though it was a public API in documentation and intelliSense.

## Expected behavior
The operations, functions, and types within the `Microsoft.Quantum.Simulation.QuantumProcessor.Extensions` namespace should not be visible outside the QSharpCore project.

## Actual behavior
![image](https://user-images.githubusercontent.com/31516/89963465-85211200-dbfc-11ea-8b3c-aa2ce97c3461.png)

**Comment by bettinaheim** [comment]
@cgranade I think that namespace shouldn't exist in the first place. For one, the namespace contains more than it needs to, and then there is a certain redundancy with what is defined in the standard libraries. It would hence be great if we could make a pass over the standard libraries, and move the operations that make sense into the core package (they can keep their namespace if that makes sense), delete what is not strictly needed from this namespace, and migrate the rest into the same namespace as the other things moved over from the standard library, and then delete this file/namespace. Does that make sense?

**Comment by cgranade** [comment]
> @cgranade I think that namespace shouldn't exist in the first place. For one, the namespace contains more than it needs to, and then there is a certain redundancy with what is defined in the standard libraries. It would hence be great if we could make a pass over the standard libraries, and move the operations that make sense into the core package (they can keep their namespace if that makes sense), delete what is not strictly needed from this namespace, and migrate the rest into the same namespace as the other things moved over from the standard library, and then delete this file/namespace. Does that make sense?

Definitely, I'm all in favor of moving things from the Microsoft.Quantum.Standard package to the QSharpCore project to help with dependency ordering. There have been several instances of that already, given that there's little to no user-facing impact (Microsoft.Quantum.Standard and QSharpCore are both brought in automatically, so the only way to notice is by explicitly disabling Microsoft.Quantum.Standard). I'm especially in favor of deleting things in this namespace that are redundant with Microsoft.Quantum.Standard functionality, but where the distinction isn't clear to users.

**Comment by bettinaheim** [comment]
@cgranade Sounds good. To you think this is something that the libraries team will have time to look into?

**Comment by cgranade** [comment]
Sure, let's see what we can do, then. At first blush, I think a reasonable starting point would be to move everything that is currently duplicated between the Microsoft.Quantum.Canon and Microsoft.Quantum.Simulation.QP.Extensions namespaces into the QSharpCore project, so that as a second step, we can refactor anything that depends on Microsoft.Quantum.Simulation.QP.Extensions to depend on the versions in Microsoft.Quantum.Canon.
